middleware/validateRoomsArr.js

```diff
@@ -0,0 +1,15 @@
+const errorMessage = require('../utils/errorMessage');
+
+const validateRoomsArr = (req, res, next) => {
+  const roomsArr = req.body;
+
+  if (!Array.isArray(roomsArr)) {
+    res.status(400).json(errorMessage.typeArr);
+  } else if (roomsArr.filter(room => !room.name).length) {
+    res.status(400).json(errorMessage.noRoomName);
+  } else {
+    next();
+  }
+};
+
+module.exports = validateRoomsArr;
```

Comment:
This could be validated in `hotelSchema` in `hotelModels.js`, with `type: Array`.

Reply:
I think that if I don't have the type check here, and someone sends a non-array,  my code will break way before it gets to the db validation -> was a way of handling the error from that essentially.

Reply:
👍

Reply:
Validation-ception 👍